feat: UTC-130: Fetching action form options on action click

mcanu opened the pull request:
<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->



https://github.com/user-attachments/assets/ee3beb12-4833-4451-a870-d480337d865f

mcanu committed:
Fetching action form options on action click

mcanu commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16033192600)

mcanu committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16033192600

mcanu committed:
Merge branch 'develop' into 'fb-utc-130'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16033192600

yyassi-heartex commented on web/libs/datamanager/src/components/DataManager/Toolbar/ActionsButton.jsx:
```diff
@@ -17,13 +17,45 @@ const injector = inject(({ store }) => ({
 }));
 
-const buildDialogContent = (text, form, formRef) => {
+// const buildDialogContent = async (text, form, formRef, store) => {
+//   const formData = await store.fetchActionForm(actionId);
+
+//   return (
+//     <Block name="dialog-content">
+//       <Elem name="text">{text}</Elem>
+//       {formData && (
+//         <Elem name="form" style={{ paddingTop: 16 }}>
+//           <Form.Builder ref={formRef} fields={formData.toJSON()} autosubmit={false} withActions={false} />
+//         </Elem>
+//       )}
+//     </Block>
+//   );
+// };
```
```suggestion
```

yyassi-heartex commented on web/libs/datamanager/src/components/DataManager/Toolbar/ActionsButton.jsx:
```diff
@@ -17,13 +17,45 @@ const injector = inject(({ store }) => ({
+  useEffect(() => {
+    if (!formData) {
+      store.fetchActionForm(action.id).then((form) => {
+        console.log("form response", form);
```
```suggestion
```

yyassi-heartex commented on web/libs/datamanager/src/components/DataManager/Toolbar/ActionsButton.jsx:
```diff
@@ -17,13 +17,45 @@ const injector = inject(({ store }) => ({
+      });
+    }
+  }, [formData]);
+
+  console.log("formData", formData);
+  console.log("formData.toJSON", formData && formData.toJSON);
+  console.log("form", form);
```
```suggestion
```

yyassi-heartex commented on web/libs/datamanager/src/components/DataManager/Toolbar/ActionsButton.jsx:
```diff
@@ -17,13 +17,45 @@ const injector = inject(({ store }) => ({
+  console.log("form", form);
+
+  const fields = formData && formData.toJSON ? formData.toJSON() : formData;
+  console.log("fields", fields);
```
```suggestion
```

yyassi-heartex reviewed (approved):
looks good overall - assuming all checks are green and selenium passes its approved

mcanu committed:
Removed prints and logs

mcanu committed:
Fetching actions on page load

mcanu committed:
Fetch actions in background

mcanu committed:
Fix api docs

mcanu committed:
Loading actions if failed to load on select click

mcanu committed:
Show loader when form is loading

mcanu commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16059094924)

mcanu committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16059094924

mcanu committed:
Merge branch 'develop' into 'fb-utc-130'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16059094924
